apps/projects/contracts/Projects.sol

```diff
@@ -23,95 +75,224 @@ contract Projects is AragonApp {
+        checkTransValueEqualsMessageValue(msg.value, _bountySizes,_tokenBounties);
+        
+        string memory ipfsHash;
+        uint standardBountyId;
+        // submit the bounty to the StandardBounties contract
+        for (uint i = 0; i < _bountySizes.length; i++) {
+            ipfsHash = substring(_ipfsAddresses, i.mul(46), i.add(1).mul(46));
+            standardBountyId = bounties.issueBounty(
+                this,                           //    address _issuer
+                _deadlines[i],                  //    uint256 _deadlines
+                ipfsHash,                       //    parse input to get ipfs hash
+                _bountySizes[i],                //    uint256 _fulfillmentAmount
+                address(0),                     //    address _arbiter
+                _tokenBounties[i],              //    bool _paysTokens
+                address(_tokenContracts[i])     //    address _tokenContract
+            );
+            // Activate the bounty so it can be fulfilled
+            bounties.activateBounty.value(_bountySizes[i])(standardBountyId, _bountySizes[i]);
+
+            //Add bounty to local registry
+            _addBounty(
+                _repoId,
+                _issueNumbers[i],
+                standardBountyId,
+                _bountySizes[i]
+            );
         }
     }
 
+///////////////////////
+// Internal functions
+///////////////////////
+
     function _addBounty(
-        bytes32 _repoID, uint256 _issueNumber, uint256 _bountySize
+        bytes32 _repoId,
+        uint256 _issueNumber,
+        uint _standardBountyId,
+        uint256 _bountySize
     ) internal 
     {
-        repos[_repoID].issues[_issueNumber] = GithubIssue(
-            _repoID,
+        repos[_repoId].issues[_issueNumber] = GithubIssue(
+            _repoId,
             _issueNumber,
             true,
-            _bountySize,
-            address(0)
+            _standardBountyId
         );
         emit BountyAdded(
-            repos[_repoID].owner,
-            repos[_repoID].repo,
+            repos[_repoId].owner,
+            repos[_repoId].repo,
             _issueNumber,
             _bountySize
         );
     }
+
+    function checkTransValueEqualsMessageValue(
+        uint256 _msgValue,
+        uint256[] _bountySizes,
+        bool[] _tokenBounties
+    ) internal pure 
+    {
+        uint256 transValueTotal = 0;
+        for (uint i = 0; i < _bountySizes.length; i++) {
+            if (!(_tokenBounties[i])) {
+                transValueTotal = transValueTotal.add(_bountySizes[i]);
+            }
+        }
+        require(_msgValue == transValueTotal, "ETH sent to cover bounties does not match bounty total");
+    }
+
+    function substring(
+        string str,
+        uint startIndex,
+        uint endIndex
+    ) internal pure returns (string)
+    {
+        // first char is at location 0
+        //IPFS addresses span from 0 (startindex) to 46 (endIndex)
```

Comment:
Since this is a generalized function I'd re-locate these comments and maybe explain this more generally here.